tests/ci_build/Dockerfile.gpu_dev_ver

```diff
@@ -0,0 +1,51 @@
+# Container to test XGBoost against dev versions of dependencies
+
+ARG CUDA_VERSION_ARG
+FROM nvidia/cuda:$CUDA_VERSION_ARG-runtime-ubuntu22.04
+ARG CUDA_VERSION_ARG
+ARG RAPIDS_VERSION_ARG
+ARG NCCL_VERSION_ARG
+
+# Environment
+ENV DEBIAN_FRONTEND noninteractive
+SHELL ["/bin/bash", "-c"]   # Use Bash as shell
+
+# Install all basic requirements
+RUN \
+    apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/3bf863cc.pub && \
+    apt-get update && \
+    apt-get install -y wget unzip bzip2 libgomp1 build-essential openjdk-8-jdk-headless && \
+    # Python
+    wget -nv -O conda.sh https://github.com/conda-forge/miniforge/releases/download/22.11.1-2/Mambaforge-22.11.1-2-Linux-x86_64.sh && \
+    bash conda.sh -b -p /opt/mambaforge
+
+ENV PATH=/opt/mambaforge/bin:$PATH
+
+# Create new Conda environment with cuDF, Dask, and cuPy
+RUN \
+    export NCCL_SHORT_VER=$(echo "$NCCL_VERSION_ARG" | cut -d "-" -f 1) && \
+    mamba create -y -n gpu_test -c rapidsai-nightly -c nvidia -c conda-forge \
+        python=3.10 "cudf>$RAPIDS_VERSION_ARG" "rmm>$RAPIDS_VERSION_ARG" cudatoolkit=$CUDA_VERSION_ARG \
+        "nccl>=${NCCL_SHORT_VER}" \
+        dask=2024.1.1 \
```

Comment:
Not sure when we can release this pin on `dask`.

Reply:
I think the dask version would be tied to the RAPIDS release version.

Reply:
+1, I think and I think you're already going to get the RAPIDS pins via having `dask-cuda` in this environment, since it depends on `rapids-dask-dependency`.

https://github.com/rapidsai/dask-cuda/blob/f1a14050ec9827c9401c6ca90617b16c358029dd/pyproject.toml#L23

https://github.com/rapidsai/rapids-dask-dependency/blob/e2938ebb44d9d28a0fc5b8c5ebcc29a57bd5b3ea/pyproject.toml#L14

I recommend removing `dask` from this list of requirements and just taking whatever versions get pulled in via `dask-cuda` and `dask-cudf`.

Reply:
Removing `dask` from the list did the trick. Thanks!